### PR TITLE
[BUGFIX] Fix workflow errors for the VL53C4X driver. Adopt the Flash driver for usage with Littlefs

### PR DESCRIPTION
**Pull request by Hoog-V**
This pull request introduces a new driver implementation for the STM VL53L4CD ranging ToF sensor, along with its integration into the build system, unit tests, and minor fixes in related files. Below is a summary of the most important changes:

### VL53L4CD Sensor Driver Implementation:
* Added a new driver for the VL53L4CD sensor, including initialization, reading, and deinitialization functions (`src/vl53l4cd/vl53l4cd.c`, `src/vl53l4cd/vl53l4cd.h`). [[1]](diffhunk://#diff-d04abd5102bca422110245985053a85031908a51819f7dc424e85c75c6831350R1-R181) [[2]](diffhunk://#diff-c7922752b4894ff784b11baf3ad7e08b9179da2086fbaab175478f31c12c5130R1-R71)
* Created a private header file defining the sensor's register map (`src/vl53l4cd/private/vl53l4cd_regs.h`).

### Build System Updates:
* Integrated the VL53L4CD driver into the build system by adding its source file to `CMakeLists.txt`.
* Added a new test executable for the VL53L4CD sensor in the test CMake configuration (`tests/CMakeLists.txt`).

### Unit Tests:
* Developed unit tests for the VL53L4CD driver to validate its behavior, including edge cases such as null parameters and clipping distance values (`tests/vl53l4cd/test_vl53l4cd.cpp`).

### Bug Fixes and Enhancements in Related Code:
* Removed redundant address calculation in `w25qxx_read` and `w25qxx_write` functions (`src/w25qxx128/w25qxx128.c`). [[1]](diffhunk://#diff-d00cff4584eafd70b9b0550f2e677980b459238edc664b94fb890cb298549820L138) [[2]](diffhunk://#diff-d00cff4584eafd70b9b0550f2e677980b459238edc664b94fb890cb298549820L204)
* Added error handling for the sector erase operation in `w25qxx_erase_sector` to increment the fault counter on failure (`src/w25qxx128/w25qxx128.c`).
* Added wait loop for  sector erase operation in `w25qxx_erase_sector` to make sure it waits for full sector erasure before proceeding. Blocking call (as that is what litlefs requires) (`src/w25qxx128/w25qxx128.c`).

### CODE REVIEW

File: src/w25qxx128/w25qxx128.c [hunk 1]
```diff
@@ -271,7 +269,11 @@ w25qxx_erase_sector (manikin_spi_memory_ctx_t *mem_ctx, uint32_t sector_number)
     }
 
     status |= w25qxx_set_address(mem_ctx, W25QXX_REG_SECTOR_ERASE, addr);
-
+    if (w25qxx_wait_while_busy(mem_ctx) != MANIKIN_STATUS_OK)
+    {
+        mem_ctx->fault_cnt++;
```
Comment (Oguz96) [inline src/w25qxx128/w25qxx128.c:282]: manikin_spi_end_transaction

File: src/w25qxx128/w25qxx128.c [hunk 2]
```diff
@@ -271,7 +269,11 @@ w25qxx_erase_sector (manikin_spi_memory_ctx_t *mem_ctx, uint32_t sector_number)
     }
 
     status |= w25qxx_set_address(mem_ctx, W25QXX_REG_SECTOR_ERASE, addr);
-
+    if (w25qxx_wait_while_busy(mem_ctx) != MANIKIN_STATUS_OK)
```
Comment (Oguz96) [inline src/w25qxx128/w25qxx128.c]: check status

File: src/w25qxx128/w25qxx128.c [hunk 3]
```diff
@@ -270,15 +268,26 @@ w25qxx_erase_sector (manikin_spi_memory_ctx_t *mem_ctx, uint32_t sector_number)
...
+    status |= manikin_spi_start_transaction(mem_ctx->spi_cs);
     if (status != MANIKIN_STATUS_OK)
     {
         manikin_spi_end_transaction(mem_ctx->spi_cs);
         mem_ctx->fault_cnt++;
         return MANIKIN_MEMORY_RESULT_ERROR;
     }
-
+    status |= w25qxx_set_address(mem_ctx, W25QXX_REG_SECTOR_ERASE, addr);
+    if ((status != MANIKIN_STATUS_OK) || (w25qxx_wait_while_busy(mem_ctx) != MANIKIN_STATUS_OK))
```
Comment (Oguz96) [inline src/w25qxx128/w25qxx128.c:279]: Overweeg om er een functie van te maken. Ik zie het een aantal keer terugkomen

**Review by Oguz96 — APPROVED** [review]
Zie suggestie
